runtime/fastly/builtins/fetch/fetch.h

```diff
@@ -0,0 +1,8 @@
+#include "headers.h"
+#include "request-response.h"
+
+namespace fastly::fetch {
+namespace {
+api::Engine *ENGINE;
```

Comment:
Could you mark this `extern` so that we don't end up with multiple definitions of `ENGINE`?

Reply:
This is supposed to be internal and duplicated - following the pattern from https://github.com/bytecodealliance/StarlingMonkey/blob/main/builtins/web/fetch/fetch_event.cpp#L18.

If there's a better pattern here perhaps we can get alignment with StarlingMonkey first? All builtins need to maintain the engine in a builtin-local variable under the current hook-based extension model.

Reply:
When defined in the `.cpp` file, that will be a constant. When defined in the header, the definition will introduce a new `ENGINE` constant for every include. I think this is probably unintentional, so I would mark this declaration `extern`, and put the real definition in `fetch.cpp`.

Reply:
The problem I was trying to solve here, is that this is shared between all the `fetch/*.cpp` files, as a kind of local state, where the fetch builtin is treated as one builtin across many files. I was under the impression that since the anonymous namespace is under the `fastly::fetch` builtin namespace we get that sharing without spilling it to other builtins in other builtin namespaces.